### ISSUE TITLE
Ground Spoilers Engage on Takeoff

### ISSUE THREAD

**Issue by legoboyvdlp** [FGMEMBERS/A320neo#9]
Everytime I takeoff, I get a master caution, and speedbrakes engage. Ground spoilers engage with no reduction of throttle.

**Comment by jan-hudec** [comment]
Hm, it's happened to me several times already, but now I wanted to try to reproduce it and it did _not_ happen.

**Comment by jan-hudec** [comment]
I am pretty sure that the bug exists, but it does not happen every time. If I just quickly start the engines, configure for take-off (flaps, spoilers armed, autobrake max) and take off, it does not happen. So it breaks by _something_ during the normal preparation. But I don't know what.

**Comment by legoboyvdlp** [comment]
Well, I always set up properly, with FGMC.
It seems to happen at V1, so likely, if you set the V1 property to 80 knots, it would break at 80 knots.

**Comment by legoboyvdlp** [comment]
Testing now.
Set FGMC with SVMI/SVMG, FL180, Flight No, Alternate, and V speeds 80 120 140.

**Comment by legoboyvdlp** [comment]
Confirmed, it happens at V1.

**Comment by jan-hudec** [comment]
Yes, it happens when V-speeds are set; for me it happens at 72 knots, not V1, though.

So I am starting at the controller and it makes absolutely no sense however. The controller checks that throttle of both engines to `/autopilot/autobrake/config/idle-throttle`, weight is on wheels and indicated speed is over 72 knots. Property browser tells me the `idle-throttle` is still 0.1 and throttles are both still 1, so the first condition shouldn't be satisfied. And I don't see any other reference to the `/controls/flight/ground-spoilers` property anywhere else either. Still, it becomes true. Only if V-speeds are set. Strange.

**Comment by jan-hudec** [comment]
Ok, so I pulled out the <del>howitzer</del><ins>gdb</ins> and traced the system. What happens is that the auto-thrust controller in `Systems/FMGC/auto-throttle.xml`:`IAS hold FMGC` tries to control the thrust always, but is later overriden by `Systems/thrust_levers.xml`:`Left`/`Right Throttle Position`. However the ground spoilers system runs between them, so it sees the thrust levers retarded. But the property browser, and FDM, both see them fully advanced, because that's where the autopilot system eventually leaves them.

Now I'll have to think about how to fix that _properly_ (it can probably be hacked quickly by rearranging the controllers, but it would be extremely fragile too).

**Comment by IAHM-COL** [comment]
Yay!
Nice backtrack!

I suggest thinking the real fix.... as you say.

(OTOH, this is already beyond my coding comprehension, so I limit myself to
the role of Cheerleader!)

**Comment by ghost** [comment]
I uploaded a fix for this Issue on my A330-200 repo. Should be the same for the A320neo. Would be nice if someone could check it out and tell me if it works for him.

I tested it and it works fine for me: Spoilers do not engange on take-off (if V-speeds are set or not). Spoilers retract on touchdown, when armed.

**Comment by FGDATA** [comment]
That sounds incredible. 

@ALL. 
I recommend the following code to test the OrangeTiger repo

```
      git checkout -b Spoiler-fix
      git pull git@github.com:OrangeTiger/A330-200.git
```

That creates a branch in your repo named Spoiler-fix, and switches to it, then it pulls OrangeTiger repo into that branch

you can return to master with

```
     git checkout master
```

and return to Spoiler-fix with

```
      git Spoiler-fix
```

It this fix works, all is needed is that OrangeTiger (or someone else) initiates a pull request from OrangeTiger repo

IH-COL

**Comment by legoboyvdlp** [comment]
Okay, plz remind me to test...

**Comment by tonghuix** [comment]
It seem not fixed yet in master.
